boa/src/syntax/parser/statement/labelled_stm/mod.rs

```diff
@@ -0,0 +1,50 @@
+use super::LabelIdentifier;
+use crate::{
+    syntax::{
+        ast::Punctuator,
+        parser::{
+            cursor::Cursor, error::ParseError, AllowAwait, AllowReturn, AllowYield, TokenParser,
+        },
+    },
+    BoaProfiler,
+};
+/// Labelled Statement Parsing
+///
+/// More information
+/// - [MDN documentation][mdn]
+/// - [ECMAScript specification][spec]
+///
+/// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/label
+/// [spec]: https://tc39.es/ecma262/#sec-labelled-statements
+#[derive(Debug, Clone, Copy)]
+pub(super) struct Label {
```

Comment:
I would keep the name of the `LabelledStatement` as in the spec. This parser would return a `LabelledStatement` node, with a label and a `Node`.